src/stream/src/executor/hash_join.rs

```diff
@@ -220,6 +230,61 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
         chunk_size: usize,
         high_join_amplification_threshold: usize,
     ) -> Self {
+        Self::new_with_cache_size(
+            ctx,
+            info,
+            input_l,
+            input_r,
+            params_l,
+            params_r,
+            null_safe,
+            output_indices,
+            cond,
+            inequality_pairs,
+            state_table_l,
+            degree_state_table_l,
+            state_table_r,
+            degree_state_table_r,
+            watermark_epoch,
+            is_append_only,
+            metrics,
+            chunk_size,
+            high_join_amplification_threshold,
+            None,
+        )
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn new_with_cache_size(
```

Comment:
It seems we could use `new_with_cache_size` implementation to replace the `new` one.

Reply:
We only need it for benchmarking purposes. `new` can implicitly decide the cache size.
It will cause more refactoring to happen, and we have to pass in redundant `None` as an argument in all instantiation sites.